### PR TITLE
[EAP7-689] Added tests for Elytron integration in mail subsystem

### PR DESCRIPTION
**Pull request by honza-kasik**
+ extended tests for mail server attributes
+ refactore - splitted one tests case to test case for mail session and to test case for mail servers
+ this depends on `ElytronIntegrationChecker` which is also defined in #243

### CODE REVIEW

**Comment by honza-kasik** [comment]
@zebrik Can you please review it when you will have some time?

**Comment by honza-kasik** [comment]
Actually, if the https://issues.jboss.org/browse/HAL-1296 lands in next build, I can re-enable ignored tests!

**Comment by rhatlapa** [comment]
There are conflicting files, please rebase.

**Comment by honza-kasik** [comment]
@rhatlapa Thanks! I resolved conflict.

**Comment by honza-kasik** [comment]
Ready to be merged if check will be OK.

**Comment by honza-kasik** [comment]
Closed by mistake. Ready to be merged!

**Comment by honza-kasik** [comment]
Resolved conflicts

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/mail/MailSubsystemOperations.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    MailSubsystemOperations(OnlineManagementClient client) {
+        this.client = client;
+    }
+
+    String createSocketBinding() throws IOException, CommandFailedException {
+        String socketBinding = "mailTestCaseSb_" + RandomStringUtils.randomAlphanumeric(6);
+        return createSocketBinding(socketBinding);
+    }
+
+    String createSocketBinding(String name) throws IOException, CommandFailedException {
```
Comment (zebrik) [inline basic/src/test/java/org/jboss/hal/testsuite/test/configuration/mail/MailSubsystemOperations.java]: Is it really necessary to have both variants of the `createSocketBinding` method? IMHO it makes the code less logic and readable. IMHO `String createSocketBinding()` should be sufficient. Just suggestion, I do not insist on such unification. WDYT?

**Review by zebrik — APPROVED** [review]

File: basic/src/test/java/org/jboss/hal/testsuite/test/configuration/mail/MailSubsystemOperations.java (hunk 1 above)
Comment (zebrik) [reply basic/src/test/java/org/jboss/hal/testsuite/test/configuration/mail/MailSubsystemOperations.java]: Needs squashing before merge.

Comment (honza-kasik) [reply basic/src/test/java/org/jboss/hal/testsuite/test/configuration/mail/MailSubsystemOperations.java]: Thanks! I removed second method, now I will squash it.

**Comment by zebrik** [comment]
LGTM. Thank you.

**Pull request merged**